katsdpcontroller/test/test_product_config.py

```diff
@@ -1,73 +1,1012 @@
+        assert_equal(override.config, {})
+        assert_equal(override.taskinfo, {})
+        assert_equal(override.host, None)
+
+
+class TestOptions:
+    """Test :class:`~.Options`."""
+
+    def test_from_config(self) -> None:
+        config = {
+            'develop': True,
+            'wrapper': 'http://test.invalid/wrapper.sh',
+            'service_overrides': {
+                'service1': {
+                    'host': 'testhost'
+                }
+            }
+        }
+        options = Options.from_config(config)
+        assert_equal(options.develop, config['develop'])
+        assert_equal(options.wrapper, config['wrapper'])
+        assert_equal(list(options.service_overrides.keys()), ['service1'])
+        assert_equal(options.service_overrides['service1'].host, 'testhost')
+
+    def test_defaults(self) -> None:
+        options = Options.from_config({})
+        assert_equal(options.develop, False)
+        assert_equal(options.wrapper, None)
+        assert_equal(options.service_overrides, {})
+
+
+class TestSimulation:
+    """Test :class:`~.Simulation`."""
+
+    def test_from_config(self) -> None:
+        config = {
+            'sources': [
+                'PKS 1934-63, radec, 19:39:25.03, -63:42:45.7, (200.0 12000.0 -11.11 7.777 -1.231)',
+                'PKS 0408-65, radec, 4:08:20.38, -65:45:09.1, (800.0 8400.0 -3.708 3.807 -0.7202)'
+            ],
+            'start_time': 1234567890.0,
+            'clock_ratio': 2.0
+        }
+        sim = Simulation.from_config(config)
+        assert_equal(sim.start_time, 1234567890.0)
+        assert_equal(sim.clock_ratio, 2.0)
+        assert_equal(len(sim.sources), 2)
+        assert_equal(sim.sources[0].name, 'PKS 1934-63')
+        assert_equal(sim.sources[1].name, 'PKS 0408-65')
+
+    def test_defaults(self) -> None:
+        sim = Simulation.from_config({})
+        assert_is_none(sim.start_time)
+        assert_equal(sim.clock_ratio, 1.0)
+        assert_equal(sim.sources, [])
+
+    def test_invalid_source(self) -> None:
+        with assert_raises_regex(ValueError, 'Invalid source 1: .* must have at least two fields'):
+            Simulation.from_config({'sources': ['blah']})
+
+
+class TestCamHttpStream:
+    """Test :class:`~.CamHttpStream`."""
+
+    def test_from_config(self) -> None:
+        config = {
+            'type': 'cam.http',
+            'url': 'http://test.invalid'
+        }
+        cam_http = CamHttpStream.from_config(Options(), 'cam_data', config, [], {})
+        assert_equal(cam_http.name, 'cam_data')
+        assert_equal(cam_http.src_streams, [])
+        assert_equal(cam_http.url, yarl.URL('http://test.invalid'))
+
+
+class TestAntennaChannelisedVoltageStream:
+    """Test :class:`~.AntennaChannelisedVoltageStream`."""
+
+    def test_from_config(self) -> None:
+        config = {
+            'type': 'cbf.antenna_channelised_voltage',
+            'url': 'spead://239.0.0.0+7:7148',
+            'antennas': ['m000', 'm001'],
```

Comment:
Any point in adding some naming diversity here (e.g. s0002) to help prevent name interpretation that might sneak in ?

Reply:
I'm pretty confident that I haven't done any such thing in the master controller code.

Reply:
My point is more about getting into the habit of using different antenna names, just because it emphasises the fact that we have different ones. I would never cast such an aspersion on your person, but lesser mortals have been known to fall into this trap....

Reply:
Ok, I've sprinkled some alternative names around the tests.